packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.tsx

```diff
@@ -35,15 +36,50 @@ export class AtomicRefineModal implements InitializableComponent {
   @State() public error!: Error;
 
   @Prop({reflect: true, mutable: true}) enabled!: boolean;
+  @Watch('enabled')
+  watchEnabled(enabled: boolean) {
+    const modalOpenedClass = 'atomic-modal-opened';
+
+    if (enabled) {
+      document.body.classList.add(modalOpenedClass);
+      this.duplicateFacetElements();
+      return;
+    }
+
+    document.body.classList.remove(modalOpenedClass);
+    this.flushFacetElements();
+  }
 
   public initialize() {
     this.breadcrumbManager = buildBreadcrumbManager(this.bindings.engine);
     this.querySummary = buildQuerySummary(this.bindings.engine);
   }
 
+  private duplicateFacetElements() {
+    const divSlot = document.createElement('div');
+    divSlot.setAttribute('slot', 'facets');
+    // TODO: order facets with facet manager
+    this.bindings.store.get('facetElements').forEach((facetElement) => {
+      const clone = facetElement.cloneNode(false) as HTMLElement;
```

Comment:
Instead of cloning the DOM element, could we create a new element using JSX, configuring the component with the necessary state?

Reply:
This is cloning the DOM tag only (deep = false) it will make the component initialize normally and connect to headless, this is not a deep copy of the current facet component

Reply:
Inserting it with JS, not inside the JSX directly as children, as we don't want a double shadow dom

Reply:
Ok, thanks for the clarifications,